### PR TITLE
bug(fix): enable eslint extension on my vscode

### PR DESCRIPTION
**Pull request by amaechi-chuks**
**What does this PR do?**
This PR sets up an eslint extension on my code editor to track my code.

Description of tasks:

Ensure that eslint is enable
Ensure that all indentation is fixed
Remove excessive comments
Remove whitespace

**How can this PR be tested**

Using any code editor of your choice
Run the code and check the following:
- Indentation
- whitespace
- excessive comments

**Link to relevant story** 👍 
#162383839

### CODE REVIEW

File: server/controllers/incident.js [hunk 1]
```diff
@@ -1,214 +1,195 @@
...
-            });
-        }
+/**
+ * API method to Post an Incident
+ * @static
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @return {object} JSON representing success message
+ * @memberof IncidentController
+*/
```
Comment (otseobande) [inline server/controllers/incident.js:16]: Indent this JsDoc a step in.

File: server/controllers/incident.js [hunk 2]
```diff
@@ -1,214 +1,195 @@
...
+
```
Comment (otseobande) [inline server/controllers/incident.js:17]: Remove this line.

File: server/controllers/incident.js [hunk 3]
```diff
@@ -1,214 +1,195 @@
...
+  /**
+  * API Get method to get all Incident
+  * @static
+  * @param {object} req - The request object
+  * @param {object} res - The response object
+  * @returns {object} JSON object representing success message
+  * @memberof IncidentController
+  */
+
```
Comment (otseobande) [inline server/controllers/incident.js:54]: Remove line

File: server/controllers/incident.js [hunk 4]
```diff
@@ -1,214 +1,195 @@
...
+  static getAllIncident(req, res) {
+    if (db.incident.length !== 0) {
+      const type = req.url.split('/')[1];
+      const type2 = type.split('s')[0];
+      const data = db.incident.filter((incidentType) => {
```
Comment (otseobande) [inline server/controllers/incident.js:59]: Make the variables used in this method more descriptive.

File: server/controllers/incident.js [hunk 5]
```diff
@@ -1,214 +1,195 @@
...
+
+  /**
+    * API PUT method to update  a single incident by location
+    * @param {object} req
+    * @param {object} res
+    * @returns {object} success message
+    * @returns {object} {object} JSON object representing success message
+    * @memberof IncidentController
+   */
+
```
Comment (otseobande) [inline server/controllers/incident.js:118]: Remove this line.

File: server/middlewares/incident.js [hunk 6]
```diff
@@ -1,337 +1,322 @@
-import db from '../models/incident'
+import db from '../models/incident';
 
 /**
  * Class representing Incident Validations
  * @class IncidentValidator
-**/
-
+* */
```
Comment (otseobande) [inline server/middlewares/incident.js:6]:
```suggestion
**/
```

File: server/middlewares/incident.js [hunk 7]
```diff
@@ -1,337 +1,322 @@
...
-     */
-    static updateByLocation(req, res, next) {
-        const { location } = req.body;
+    req.body.createdOn = createdOn;
+    req.body.createdBy = createdBy;
+    req.body.type = type;
+    req.body.location = location;
+    req.body.status = status;
+    req.body.imageUrl = imageUrl;
+    req.body.videoUrl = videoUrl;
```
Comment (otseobande) [inline server/middlewares/incident.js:243]: Please explain the assignments happening here

File: server/middlewares/user.js [hunk 8]
```diff
@@ -3,333 +3,317 @@ import db from '../models/incident';
...
-        }
-        req.body.foundUser = foundUser;
-        req.body.password = password;
-        return next();
+/**
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @param {function} next - Calls the next function
+ * @returns {object} JSON representing the failure message
+*/
```
Comment (otseobande) [inline server/middlewares/user.js:13]:
```suggestion
 */
```

File: server/middlewares/user.js [hunk 9]
```diff
@@ -3,333 +3,317 @@ import db from '../models/incident';
...
+  static signUpValidator(req, res, next) {
+    // destructure user object property
```
Comment (otseobande) [inline server/middlewares/user.js:15]: This comment does not add value to your codebase

File: server/middlewares/user.js [hunk 10]
```diff
@@ -3,333 +3,317 @@ import db from '../models/incident';
...
+    }
+
+    req.body.firstName = firstName;
+    req.body.lastName = lastName;
+    req.body.otherNames = otherNames;
+    req.body.email = email;
+    req.body.password = password;
+    req.body.confirmPassword = confirmPassword;
+    req.body.phoneNumber = phoneNumber;
+    req.body.username = username;
```
Comment (otseobande) [inline server/middlewares/user.js:250]: I don't understand your assignment to `req.body`

File: server/test/incident.js [hunk 11]
```diff
@@ -21,604 +23,500 @@ const notFound = `/api/v1/red-flag/${notFoundRedFlagId}`;
...
+    });
+    it('Should return 400 status code for incident date not a string', (done) => {
+      chai.request(app)
+        .post(url)
+        .send(inValidIncidentDate[1])
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          expect(res.body.status).to.equal(400);
+          res.body.message.should.be.a('string');
```
Comment (otseobande) [inline server/test/incident.js:48]: Please test that the content of the message matches requirements.

**Review by otseobande — CHANGES_REQUESTED** [review]
Good job implementing some of the feedback I gave yesterday. Please address the comments I left below.

File: server/middlewares/incident.js (hunk 7 above)
Comment (amaechi-chuks) [reply server/middlewares/incident.js:243]: I found out that the assignment is not necessary. I noticed it when I console log it to see the output I am getting. I now realized that after destructuring it at the top level, there's no need reassigning it again. I have removed it from the codebase.

File: server/middlewares/user.js (hunk 10 above)
Comment (amaechi-chuks) [reply server/middlewares/user.js:250]: Acknowledged and fixed

File: server/test/incident.js (hunk 11 above)
Comment (amaechi-chuks) [reply server/test/incident.js:48]: Acknowledged and fixed

File: server/middlewares/user.js (hunk 9 above)
Comment (amaechi-chuks) [reply server/middlewares/user.js:15]: Acknowledged and fixed

File: server/middlewares/user.js (hunk 8 above)
Comment (amaechi-chuks) [reply server/middlewares/user.js:13]: Acknowledged and fixed

File: server/controllers/incident.js (hunk 1 above)
Comment (amaechi-chuks) [reply server/controllers/incident.js:16]: Acknowledged and fixed

File: server/controllers/incident.js (hunk 3 above)
Comment (amaechi-chuks) [reply server/controllers/incident.js:54]: Acknowledged and fixed

File: server/controllers/incident.js (hunk 2 above)
Comment (amaechi-chuks) [reply server/controllers/incident.js:17]: Acknowledged and fixed

File: server/controllers/incident.js (hunk 4 above)
Comment (amaechi-chuks) [reply server/controllers/incident.js:59]: Acknowledged and fixed

File: server/controllers/incident.js (hunk 5 above)
Comment (amaechi-chuks) [reply server/controllers/incident.js:118]: Acknowledged and fixed

**Pull request closed without merging**